smart-contracts/for-eth-devs/accounts.md

```diff
@@ -83,9 +120,25 @@ The registration process is implemented through the [`map`](https://paritytech.g
 
 The fallback mechanism is integrated into the [`to_account_id`](https://paritytech.github.io/polkadot-sdk/master/pallet_revive/trait.AddressMapper.html#tymethod.to_account_id){target=\_blank} function. It provides a safety net for address conversion by:
 
-- First, attempting to retrieve stored mapping data.
-- Falling back to the default conversion method if no mapping exists.
-- Maintaining consistency in address representation.
+- First, attempting to retrieve stored mapping data from [`OriginalAccount`](https://paritytech.github.io/polkadot-sdk/master/pallet_revive/pallet/storage_types/struct.OriginalAccount.html){target=\_blank} storage
+- Falling back to the default conversion method (Keccak-256 hash) if no explicit mapping exists
+- Maintaining consistency in address representation across the system
+
+### How Fallback Works with Unmapped Accounts
+
+When an unmapped 32-byte Polkadot account needs to be represented as a 20-byte Ethereum address, the system:
+
+1. **Checks for explicit mapping**: First looks in the `OriginalAccount` storage to see if the account has been explicitly mapped via `map_account`
+2. **Applies fallback conversion**: If no mapping exists, automatically converts the 32-byte account to a 20-byte address by:
+   - Hashing the full 32-byte account with Keccak-256
+   - Taking the last 20 bytes of the resulting hash
+3. **Uses the derived address**: This fallback address can receive funds, but the account owner cannot spend those funds through Ethereum-compatible interfaces without explicit mapping
+
+**Important Considerations:**
+
+- The fallback mechanism is **one-way for unmapped accounts** - while you can derive the 20-byte address from a 32-byte account, you cannot recover the original 32-byte account from the 20-byte hash without the stored mapping
+- Funds sent to a fallback address of an unmapped account are not lost, but require explicit mapping to be accessible through Ethereum tools
+- For security and usability, it's recommended to establish explicit mappings rather than relying on fallback addresses for accounts that need Ethereum compatibility
```

Comment:
```suggestion
- The fallback mechanism is **one-way for unmapped accounts**. While you can derive the 20-byte address from a 32-byte account, you cannot recover the original 32-byte account from the 20-byte hash without the stored mapping.
- Funds sent to a fallback address of an unmapped account are not lost, but require explicit mapping to be accessible through Ethereum tools.
- For security and usability, it's recommended to establish explicit mappings rather than relying on fallback addresses for accounts that need Ethereum compatibility.
```